source/shared/cpp/ObjectModel/MarkDownParser.cpp

```diff
@@ -0,0 +1,430 @@
+#include <vector>
+#include <iomanip>
+#include <iostream>
+#include "MarkDownParser.h"
+
+using namespace AdaptiveCards;
+
+MarkDownParser::MarkDownParser (const std::string &txt) : m_currentDelimiterType (Init), 
+    m_currentWordIndex(0), m_lookBehind(Init), m_delimiterCnts(0), m_EmphasisState (EmphasisNotDetected), 
+    m_checkLookAhead(false), m_checkIntraWord(false), m_isRightEmphasisDetecting(false), m_text(txt), 
+    m_textBegin(m_text.begin()), m_curPos(m_text.begin()), m_textEnd(m_text.end()), m_LeftEmphasisDetecting(0),
+    m_leftLookUpTable (), m_rightLookUpTable (), m_tokenizedString(1, "")
+{
+}
+
+int MarkDownParser::adjustDelimsCntsAndMetaData(int leftOver, std::list<Emphasis>::iterator left_itr, 
```

Comment:
>Cnts [](start = 32, length = 4)

Please don't use abbreviations in function or variable names.  It makes the code harder to read.